### PR TITLE
Huge job

### PR DESCRIPTION
**Pull request by manufino**
I've overhauled the entire settings system, and  integrated the FlatLaf theme system; there's still a bit missing, but everything is already functional.

There are quite a few changes, and I hope everything works on the other platforms as well.

I've already modified the scripts to integrate the new library. I've only tested it on Windows, but they should work for the "compile" on Linux as well.

@DarwinNE , try comparing the loading speed of the options window with the previous version.

### CODE REVIEW

File: src/fidocadj/dialogs/DialogOptions.java [hunk 1]
```diff
@@ -557,90 +563,287 @@ private JPanel createPCBsizePanel()
...
-        extensionsPanel.setOpaque(false);
 
-        extStrict_CB = new JCheckBox(Globals.messages.getString(
-                "strict_FC_comp"));
-        extStrict_CB.setSelected(extStrict);
-        extStrict_CB.setOpaque(false);
+        // Groupbox "Theme Management"
+        JPanel themeManagementPanel = new JPanel();
+        themeManagementPanel.setBorder(
+                BorderFactory.createTitledBorder("Gestione tema"));
```
Comment (DarwinNE) [inline src/fidocadj/dialogs/DialogOptions.java]: Is that normal that resources in Italian appear here?

File: src/fidocadj/FidoFrame.java [hunk 2]
```diff
@@ -731,107 +714,69 @@ public FidoFrame createNewInstance()
...
-            preferences.put("pcbPadStyle", ""+ae.pcbPadStyle);
-            preferences.put("pcbPadDrill", ""+ae.pcbPadDrill);
-            preferences.put("pcbThickness", ""+ae.pcbThickness);
-            preferences.put("SHIFT_CP", cpa.getShiftCopyPaste()?"true":"false");
-
-        }
-        if(!libDirectory.equals(oldDirectory)) {
+        AddElements ae = eea.getAddElements();
+
+        // Creare e visualizzare la finestra di dialogo per le impostazioni
```
Comment (DarwinNE) [inline src/fidocadj/FidoFrame.java:742]: Do not leave comments in Italian

File: src/fidocadj/dialogs/settings/SettingsManager.java [hunk 3]
```diff
@@ -134,12 +136,15 @@ public int getInt(String key, int defaultValue)
      */
     public double getDouble(String key, double defaultValue)
     {
-        if (settings.containsKey(key)) {
-            return (double) settings.get(key);
+        String value = get(key, String.valueOf(defaultValue));
+        try {
+            return Double.parseDouble(value);
+        } catch (NumberFormatException e) {
+            // Se la conversione fallisce, restituisce il valore di default
```
Comment (DarwinNE) [inline src/fidocadj/dialogs/settings/SettingsManager.java:143]: Do not leave the comment in Italian.

File: bin/MessagesBundle_cs.properties [hunk 4]
```diff
@@ -218,6 +218,21 @@ lib_in_dir=Knihovny v adresáři
 Origin=Počátek
 Choose_browser = Výběr prohlížeče
 
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_cs.properties:221]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on it.

File: bin/MessagesBundle_de.properties [hunk 5]
```diff
@@ -115,6 +115,22 @@ pcbline_width = Leiterbahnbreite:
 PCBpad = PCB Pad
 pcbpad_height = Höhe Pad:
 pcbpad_intw = Durchmesser Pad Bohrung:
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_de.properties:119]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on it.

File: bin/MessagesBundle_el.properties [hunk 6]
```diff
@@ -115,6 +115,22 @@ Restart= Εκκίνηση
 Drawing= Σχεδίαση
 FidoCad= Επεκτάσεις FidoCadJ
 pcbline_width = Πλάτος γραμμής PCB:
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_el.properties:119]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on it.

File: bin/MessagesBundle_en.properties [hunk 7]
```diff
@@ -53,14 +53,14 @@ Circ_opt = Options
 Macro_origin = Show macro origin
 View = View
 Export_completed = Export has been done.
-Export_error = Error during exporting: 
+Export_error = Error during exporting:
```
Comment (DarwinNE) [inline bin/MessagesBundle_en.properties:56]: The spaces play a role, here for the spacing. I would not remove them, unless you solved the problem of the spacing in the layout of the dialog.

File: bin/MessagesBundle_en.properties [hunk 8]
```diff
@@ -114,17 +114,33 @@ extFCJ_c=Activate FidoCadJ extension while copying elements
 Restart=Startup
 Drawing=Drawing
 FidoCad=FidoCadJ extensions
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_en.properties:118]: Put all the resources at the end of the file, so to keep roughly the same order in all files.

File: bin/MessagesBundle_es.properties [hunk 9]
```diff
@@ -114,6 +114,22 @@ extFCJ_c=Activar extensión de FidoCadJ mientras se copian elementos
 Restart=Reiniciar
 Drawing=Dibujar
 FidoCad=Extensiones FidoCadJ
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_es.properties:118]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on them.

File: bin/MessagesBundle_fr.properties [hunk 10]
```diff
@@ -115,6 +115,22 @@ Restart=Démarrage
 Drawing=Dessin
 FidoCad=Extensions FidoCadJ
 pcbline_width = Largeur piste C.I. :
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_fr.properties:119]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on them. (PS I will translate them, so it is not a big deal, but that keeps the order approximately the same among files)

File: bin/MessagesBundle_it.properties [hunk 11]
```diff
@@ -53,14 +53,14 @@ Circ_opt = Opzioni
...
-Export_error = Errore durante l'esportazione: 
+Export_error = Errore durante l'esportazione:
 PDF_Font_error = Certi font non sono stati esportati. Utilizzare Helvetica, Courier, Times or Symbol.
 Illegal_filename = Nome del file non valido.
 File_not_found = File non trovato.
 Print_uncomplete = Impossibile terminare la stampa.
-Open_error = Errore di accesso al file: 
-Save_error = Errore durante il salvataggio:  
-Time = Tempo impiegato 
+Open_error = Errore di accesso al file:
```
Comment (DarwinNE) [inline bin/MessagesBundle_it.properties:61]: Same here, check if the space is required for the spacing. If not, it is OK to remove it.

File: bin/MessagesBundle_ja.properties [hunk 12]
```diff
@@ -115,6 +115,22 @@ Restart=起動
 Drawing=描画
 FidoCad=FidoCadJ拡張
 pcbline_width = PCB配線幅:
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_ja.properties:119]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on them.

File: bin/MessagesBundle_nl.properties [hunk 13]
```diff
@@ -115,6 +115,22 @@ Restart = Herstarten
 Drawing = Tekening
 FidoCad = FidoCadJ extenties
 pcbline_width = PCB lijnbreedte:
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_nl.properties:119]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on them.

File: bin/MessagesBundle_zh.properties [hunk 14]
```diff
@@ -216,6 +216,22 @@ Copy_split = Copy, split non standard macros
 Save_split = Save, split non standard macros
 Font_size_invalid=The font size of the text associated to graphic elements should be positive. 
 lib_in_dir=Libraries in directory
+
+Theme_management=Theme and color management
```
Comment (DarwinNE) [inline bin/MessagesBundle_zh.properties:220]: Put all the untranslated resources at the end of the file. They are easier to find for those who would like to work on them.

File: src/fidocadj/FidoMain.java [hunk 15]
```diff
@@ -395,65 +395,61 @@ public CreateSwingInterface()
...
+        boolean isDarkTheme = theme.equals("dark");
 
-            SettingsManager settingsManager = 
-                                new SettingsManager(AccessResources.class);
+        try {
+            if (enableCustomThemes) {
+                applyTheme(isLightTheme, isDarkTheme);
+            }
+        } catch (Exception e) {
+            System.out.println("Failed to apply theme. Falling back to default.");
```
Comment (DarwinNE) [inline src/fidocadj/FidoMain.java:423]: Should we write on the error stream for a message like this?

**Review by DarwinNE — CHANGES_REQUESTED** [review]
Amazing work! Can you just move the new resources in the resource files towards the end of the file? It makes things easier to locate them when someone wants to translate the new resources.

File: src/fidocadj/dialogs/DialogOptions.java (hunk 1 above)
Comment (manufino) [reply src/fidocadj/dialogs/DialogOptions.java]: It was fixed in the subsequent commits... **(Outdated)**

File: src/fidocadj/FidoFrame.java (hunk 2 above)
Comment (manufino) [reply src/fidocadj/FidoFrame.java:742]: I'll fix them right away, just some temporary comments that slipped through...

File: src/fidocadj/dialogs/settings/SettingsManager.java (hunk 3 above)
Comment (manufino) [reply src/fidocadj/dialogs/settings/SettingsManager.java:143]: I'll fix them right away, just some temporary comments that slipped through...

File: bin/MessagesBundle_cs.properties (hunk 4 above)
Comment (manufino) [reply bin/MessagesBundle_cs.properties:221]: ok

File: bin/MessagesBundle_de.properties (hunk 5 above)
Comment (manufino) [reply bin/MessagesBundle_de.properties:119]: ok

File: src/fidocadj/dialogs/DialogOptions.java (hunk 1 above)
Comment (DarwinNE) [reply src/fidocadj/dialogs/DialogOptions.java]: Ok, sorry, I was looking at the commits and reading the code for each modification. Sometimes the review system here is not crystal clear to follow.

**Comment by manufino** [comment]
I'm not sure why all those spaces were added to the translation files; maybe Notepad++ is automatically changing something (I'll need to check). Anyway, I downloaded the clean files from your branch and added what was needed at the end using Windows Notepad without touching anything else.

File: src/fidocadj/dialogs/DialogOptions.java (hunk 1 above)
Comment (manufino) [reply src/fidocadj/dialogs/DialogOptions.java]: No problem, you have quite a bit to review today! :-)

File: bin/MessagesBundle_nl.properties (hunk 13 above)
Comment (manufino) [reply bin/MessagesBundle_nl.properties]: Maybe so, but I'd still like to review the entire exception handling system if you agree. Take a look here: https://github.com/DarwinNE/FidoCadJ/discussions/254

File: src/fidocadj/FidoMain.java (hunk 15 above)
Comment (manufino) [reply src/fidocadj/FidoMain.java:423]: Maybe so, but I'd still like to review the entire exception handling system if you agree. Take a look here: https://github.com/DarwinNE/FidoCadJ/discussions/254

**Pull request merged**